src/main/java/top/theillusivec4/curios/api/type/capability/IItemCurio.java

```diff
@@ -0,0 +1,343 @@
+/*
+ * Copyright (c) 2018-2020 C4
+ *
+ * This file is part of Curios, a mod made for Minecraft.
+ *
+ * Curios is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Curios is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with Curios.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package top.theillusivec4.curios.api.type.capability;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+import com.mojang.blaze3d.matrix.MatrixStack;
+import java.util.List;
+import javax.annotation.Nonnull;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.IRenderTypeBuffer;
+import net.minecraft.client.renderer.entity.EntityRenderer;
+import net.minecraft.client.renderer.entity.LivingRenderer;
+import net.minecraft.client.renderer.entity.model.BipedModel;
+import net.minecraft.client.renderer.entity.model.EntityModel;
+import net.minecraft.client.renderer.model.ModelRenderer;
+import net.minecraft.enchantment.EnchantmentHelper;
+import net.minecraft.enchantment.Enchantments;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.ai.attributes.Attribute;
+import net.minecraft.entity.ai.attributes.AttributeModifier;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.particles.ItemParticleData;
+import net.minecraft.particles.ParticleTypes;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.vector.Vector3d;
+import net.minecraft.util.math.vector.Vector3f;
+import net.minecraft.util.text.ITextComponent;
+import top.theillusivec4.curios.api.CuriosCapability;
+import top.theillusivec4.curios.api.type.ISlotType;
+import top.theillusivec4.curios.api.type.capability.ICurio.DropRule;
+import net.minecraft.item.Item;
+
+/**
+ * Designed to be directly implemented on {@link Item} objects.<br/><br/>
+ * Curios will automatically create and attach {@link ICurio} capability to any ItemStacks that contain items
+ * implementing this interface, redirecting all calls made on such capability to respective methods here.
+ * @author Extegral
+ */
+
+public interface IItemCurio {
+
+  /**
+   * Default instance of {@link ICurio}, where all calls are redirected by default methods
+   * of this interface to avoid needlessly copying over code from there.
+   */
+
+  public static final ICurio defaultInstance = new ICurio(){};
```

Comment:
Is there any reason why `IItemCurio` does not simply extend the `ICurio` interface instead of delegating everything through this default instance?

Reply:
I thought this is obvious, but alright. The reason is because most methods are replaced with stack-sensitive versions in `IItemCurio`, and extending `ICurio` means also inheriting non-sensitive versions, which we will then need to either handle alongside with sensitive ones somehow or override solely to mark them as deprecated and tell user that they should use stack sensitive ones. Either way extending `ICurio` would pollute the resulting interface with a ton of unnecessary methods which we cannot override the visibility of, which I see no real reason to go for.